package.json

```diff
@@ -21,11 +21,12 @@
   "license": "MIT",
   "author": "Younes Aassila",
   "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
   "files": [
     "./dist/**"
   ],
   "scripts": {
-    "build": "tsup src/index.ts --no-splitting --minify",
+    "build": "tsup src/index.ts --cjsInterop --dts --splitting --minify",
```

Comment:
I needed to enable `--cjsInterop` and change the export from `module.exports =` to `export default` to get the dts file not to be empty. Unfortunately there is [a bug](https://github.com/egoist/tsup/issues/572#issuecomment-1680153238) which means that this also requires `--splitting`.